NBitcoin.Tests/ProtocolTests.cs

```diff
@@ -262,11 +261,13 @@ public void CanProcessAddressGossip()
 
 				var nodeClient = node.CreateNodeClient();
 				nodeClient.VersionHandshake();
-				using (var list = nodeClient.CreateListener().Where(m => m.Message.Payload is AddrPayload))
+				AddrV2Payload addr;

```

Comment:
What is the reason to (effectively) remove that original test? `AddrPayload` should still work, shouldn't it?

Reply:
It is because from now NBitcoin signals its preference for addrv2 during the handshake and then the node will send `addrv2` messages instead of `addr` messages.

Reply:
Should then a test for `addr` message be added? Or should the message be removed?

Reply:
The problem is that this test uses the `AddPeerAddress` rpc call that is only available in v0.21.0+ so, it is not possible to have this same test for previous versions. In fact, I added this test in a previous PR. How do you think a test like this could be implemented?